app/uk/gov/hmrc/servicedependencies/persistence/SlugInfoRepositoryBase.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.servicedependencies.persistence
+
+import com.google.inject.{Inject, Singleton}
+import org.mongodb.scala.model.Indexes.{ascending, hashed}
+import org.mongodb.scala.model.{IndexModel, IndexOptions}
+import play.api.Logger
+import play.api.libs.json.Format
+import uk.gov.hmrc.mongo.component.MongoComponent
+import uk.gov.hmrc.mongo.play.PlayMongoCollection
+
+import scala.concurrent.ExecutionContext
+import scala.reflect.ClassTag
+
+@Singleton
+abstract class SlugInfoRepositoryBase[A: ClassTag] @Inject()(mongo: MongoComponent, domainFormat: Format[A])(implicit ec: ExecutionContext)
+    extends PlayMongoCollection[A](
+      collectionName = "slugInfos",
```

Comment:
My understanding is that there is just one MongoCollection called `slugInfos`. We've made this abstract because potentially there is a union of JSObjects that are stored. We create multiple versions of SlugInfoRepositoryBase subclasses depending on the type of object we are interested in.

I wonder if this is a limitation of the library that we might want to explore? ie being able to provide a Seq() of formats instead of just one format? We can then inject this `SlugInfoRepository` instead of extending it.

Reply:
Yeah, this is an interesting one, the SlugInfoRepository has exposes multiple different types, that are all driven from the same underlying collection. Was chatting with Colin about this, for now we've gone with separate PlayMongoCollection's that each use the relevant codec on the underlying repo, and this base class is to avoid redefining the indexes for each, which is one quirk. Is worth a discussion, as it could be a good use-case to support in the library